bundle/artifacts/whl/infer.go

```diff
@@ -18,7 +18,7 @@ func (m *infer) Apply(ctx context.Context, b *bundle.Bundle) error {
 	if err != nil {
 		return err
 	}
-	artifact.BuildCommand = fmt.Sprintf("%s setup.py bdist_wheel", py)
+	artifact.BuildCommand = fmt.Sprintf("%s -m pip install setuptools wheel && %s setup.py bdist_wheel", py, py)
```

Comment:
This potentially taints the global environment.

Not convinced we should do this instead of proper venv management.

Reply:
Also keep in mind that this will try to reach the internet, which is not going to be possible in all environments, and if it isn't, the command will fail.

Reply:
> Not convinced we should do this instead of proper venv management.

I'm not suggesting to do this _instead_ of proper venv management. We still need to do that work as called out in the description above.

The present fix is simply a stopgap to prevent the unactionable error that can appear when users type`databricks bundle deploy`.

> Also keep in mind that this will try to reach the internet, which is not going to be possible in all environments, and if it isn't, the command will fail.

That's a fair point! What if we add some code to detect if wheel/setuptools are installed and show a better error? Until we have the longer-term fixes from the PR description?

Reply:
@pietern thoughts?
> What if we add some code to detect if wheel/setuptools are installed and show a better error? Until we have the longer-term fixes from the PR description?

Reply:
That works -- do you have an idea on how to do this in a portable manner?

E.g. we could run the following as part (or prior to) the build command:
```
python3 -c 'import setuptools; import wheel;'
```

Exit status will tell us if they are available.

Reply:
The second-order question is how we bubble this up as an actionable error without hard-coding this?

We could for example have a pre-build command that returns its stdout as error message.

Reply:
@andrewnester  @pietern I had a brief look at reporting an error but it's basically not possible right now due to a limitation of `artifact.go`.

What I'd like to do is something like

```
artifact.BuildCommand = fmt.Sprintf(`
%s -c '
import sys
try:
	import setuptools, wheel
except:
	print("ERROR: insert helpful error here")
	sys.exit(1)
' &&
%s setup.py bdist_wheel`, py, py)
```

but it turns out that's actually not possible right now given how artifact.go tries to "parse" the build script (https://github.com/databricks/cli/blob/main/bundle/config/artifact.go#L56-L58). This parsing makes it impossible to pass a multi-word argument to `python -c`. 

I think we need to change artifact.go to simply pass on the string to sh / cmd.exe.